File: kino-frontend/src/app/core/models/media-details-dto.ts
import { CastDto } from './cast-dto';
import { KeywordDto } from './keyword-dto';
import { MediaCardDto, MediaType } from './media-card.dto';
import { VideoDto } from './video-dto';

export interface MediaDetailsDto{
  id: number
  mediaType: MediaType;
  title: string;
  tagline: string | null;
  overview: string | null;

  posterUrl: string | null;
  backdropUrl: string | null;

  rating: string | null;
  date: string | null;
  genres: string[];
  runtimeMinutes: number | null;
  status: string | null;
  originalLanguage: string | null;
  creator: string | null;
  directors: string[];
  cast: CastDto[];
  videos: VideoDto[];
  keywords: KeywordDto[];
  recommendations: MediaCardDto[];
}

File: kino-backend/src/main/java/mk/ukim/finki/wp/kino/dto/tmdb/details/MediaDetailsDto.java
package mk.ukim.finki.wp.kino.dto.tmdb.details;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class MediaDetailsDto {
    private Long id;
    private String mediaType;
    private String title;
    private String tagline;
    private String overview;
    private String posterUrl;
    private String backdropUrl;
    private Double rating;
    private String date;
    private List<String> genres;
    private Integer runtimeMinutes;
    private String status;
    private String originalLanguage;
    private String creator;
    private List<CastDto> cast;
    private List<String> directors;
    private List<VideoDto> videos;

}
